### PR TITLE
Fixed #25254 -- Added attribute in JsonResponse to pass kwargs to json.dumps

### PR DESCRIPTION
**Pull request by darkryder**

### CODE REVIEW

File: docs/ref/request-response.txt [hunk 1]
```diff
@@ -931,6 +931,9 @@ JsonResponse objects
    serialize the data. See :ref:`JSON serialization
    <serialization-formats-json>` for more details about this serializer.
 
+   The ``json_dumps_params`` is a dictionary used to pass named arguments to
```
Comment (timgraham) [inline docs/ref/request-response.txt]: Need a `.. versionchanged:: 1.9` annotation (see elsewhere for an example) and release notes.

File: django/http/response.py [hunk 2]
```diff
@@ -492,12 +492,15 @@ class JsonResponse(HttpResponse):
       ``django.core.serializers.json.DjangoJSONEncoder``.
     :param safe: Controls if only ``dict`` objects may be serialized. Defaults
       to ``True``.
+    :param json_dumps_params: A dictionary of kwargs that would be passed
+      to json.dumps
     """
 
-    def __init__(self, data, encoder=DjangoJSONEncoder, safe=True, **kwargs):
+    def __init__(self, data, encoder=DjangoJSONEncoder,
+                    json_dumps_params={}, safe=True, **kwargs):
```
Comment (timgraham) [inline django/http/response.py]: Don't use a mutable default: `{}`. Should default to `None` and then add :
```
if json_dumps_params is None:
    json_dumps_params={}
```

File: django/http/response.py [hunk 3]
```diff
@@ -492,12 +492,15 @@ class JsonResponse(HttpResponse):
       ``django.core.serializers.json.DjangoJSONEncoder``.
     :param safe: Controls if only ``dict`` objects may be serialized. Defaults
       to ``True``.
+    :param json_dumps_params: A dictionary of kwargs that would be passed
```
Comment (timgraham) [inline django/http/response.py]: would be -> is

File: django/http/response.py [hunk 4]
```diff
@@ -492,12 +492,15 @@ class JsonResponse(HttpResponse):
...
+      to json.dumps
```
Comment (timgraham) [inline django/http/response.py]: json.dumps(). (add parens and period)

File: django/http/response.py (hunk 2 above)
Comment (timgraham) [inline django/http/response.py]: `json_dumps_params` should be after `safe` (reordering keywords could be backwards incompatible if passing them by argument).

File: docs/ref/request-response.txt [hunk 5]
```diff
@@ -914,7 +914,7 @@ JsonResponse objects
 
 .. class:: JsonResponse
 
-.. method:: JsonResponse.__init__(data, encoder=DjangoJSONEncoder, safe=True, **kwargs)
+.. method:: JsonResponse.__init__(data, encoder=DjangoJSONEncoder, safe=True, json_dumps_params=None **kwargs)
```
Comment (claudep) [inline docs/ref/request-response.txt]: Missing comma

File: docs/releases/1.9.txt [hunk 6]
```diff
@@ -545,6 +545,9 @@ Requests and Responses
 * Added :meth:`HttpRequest.get_port() <django.http.HttpRequest.get_port>` to
   fetch the originating port of the request.
 
+* Added :attr:`json_dumps_params` to :class:`~django.http.JsonResponse`
```
Comment (timgraham) [inline docs/releases/1.9.txt]: attr is incorrect (broken link and not an attribute of the class).
```
* Added the ``json_dumps_params`` parameter to
  :class:`~django.http.JsonResponse` to allow passing keyword arguments to the
  ``json.dumps()`` call used to generate the response.
```

File: docs/ref/request-response.txt [hunk 7]
```diff
@@ -936,6 +936,11 @@ JsonResponse objects
    are allowed). If ``safe`` is ``True`` and a non-``dict`` object is passed as
    the first argument, a :exc:`TypeError` will be raised.
 
+   .. versionchanged:: 1.9
```
Comment (timgraham) [inline docs/ref/request-response.txt]: This annotation should be "self-contained" as described in https://docs.djangoproject.com/en/1.8/internals/contributing/writing-documentation/#documenting-new-features.

**Comment by darkryder** [comment]
Hey! Do you need a rebase?

**Comment by timgraham** [comment]
Doesn't hurt. You may also squash commits.

**Comment by timgraham** [comment]
merged in d0bd5330432e1dda519ebd89606bd0980a36dcb4, thanks!

**Pull request closed without merging**